ngraph/frontend/paddlepaddle/CMakeLists.txt

```diff
@@ -0,0 +1,103 @@
+file(GLOB proto_files ${CMAKE_CURRENT_SOURCE_DIR}/src/proto/*.proto)
+
+if(CMAKE_CROSSCOMPILING)
+    set(PDPD_PROTOC_EXECUTABLE ${SYSTEM_PROTOC})
+else()
+    set(PDPD_PROTOC_EXECUTABLE $<TARGET_FILE:protoc>)
+endif()
+
+message("PDPD_PROTOC_EXECUTABLE is [${PDPD_PROTOC_EXECUTABLE}]")
+
+foreach(INFILE ${proto_files})
+    get_filename_component(FILE_DIR ${INFILE} DIRECTORY)
+    get_filename_component(FILE_WE ${INFILE} NAME_WE)
+    set(OUTPUT_PB_SRC ${CMAKE_CURRENT_BINARY_DIR}/${FILE_WE}.pb.cc)
+    set(OUTPUT_PB_HEADER ${CMAKE_CURRENT_BINARY_DIR}/${FILE_WE}.pb.h)
+    set(GENERATED_PROTO ${INFILE})
+    add_custom_command(
+            OUTPUT "${OUTPUT_PB_SRC}" "${OUTPUT_PB_HEADER}"
+            COMMAND ${PDPD_PROTOC_EXECUTABLE} ARGS --cpp_out ${CMAKE_CURRENT_BINARY_DIR} -I ${FILE_DIR} ${FILE_WE}.proto
+            DEPENDS ${PDPD_PROTOC_EXECUTABLE} ${GENERATED_PROTO}
+            COMMENT "Running C++ protocol buffer compiler (${PDPD_PROTOC_EXECUTABLE}) on ${GENERATED_PROTO}"
+            VERBATIM)
+    list(APPEND PROTO_SRCS "${OUTPUT_PB_SRC}")
+    list(APPEND PROTO_HDRS "${OUTPUT_PB_HEADER}")
+endforeach()
+
+add_custom_target(${TARGET_NAME}_proto DEPENDS ${PROTO_SRCS} ${PROTO_HDRS})
+
+set_source_files_properties(${PROTO_SRCS} ${PROTO_HDRS} PROPERTIES GENERATED TRUE)
+
+# Disable all warnings for generated code
+set_source_files_properties(${PROTO_SRCS} ${PROTO_HDRS} PROPERTIES COMPILE_FLAGS -w)
+
+include_directories(${Protobuf_INCLUDE_DIRS} ${${TARGET_NAME}_INCLUDE_DIR})
+
+# Create shared library
+add_library(${TARGET_NAME} SHARED  ${LIBRARY_SRC} ${LIBRARY_HEADERS} ${LIBRARY_PUBLIC_HEADERS} ${PROTO_SRCS} ${PROTO_HDRS})
+
+add_dependencies(${TARGET_NAME} libprotobuf paddlepaddle_ngraph_frontend_proto)
+
+if(NOT WIN32)
+    # Addition dependency on static libprotobuf for non-Windows
+    add_dependencies(${TARGET_NAME} libprotobuf_static)
+endif()
+
+target_include_directories(${TARGET_NAME}
+        PRIVATE
+            ${CMAKE_CURRENT_SOURCE_DIR}/src
+            ${FRONTEND_INCLUDE_PATH}
+            ${CMAKE_CURRENT_BINARY_DIR})
+
+if(COMMAND ie_add_vs_version_file)
+    ie_add_vs_version_file(NAME ${TARGET_NAME}
+                           FILEDESCRIPTION "FrontEnd to load and convert PaddlePaddle file format")
+endif()
+
+if(WIN32)
+    target_link_libraries(${TARGET_NAME} PRIVATE libprotobuf PUBLIC ngraph PRIVATE ngraph::builder)
+else()
+    target_link_libraries(${TARGET_NAME} PRIVATE ${PROTOBUF_STATIC_LIB_OUTPUT} PUBLIC ngraph PRIVATE ngraph::builder)
+endif()
+
+target_link_libraries(${TARGET_NAME} PRIVATE frontend_manager)
+
+add_clang_format_target(${TARGET_NAME}_clang FOR_TARGETS ${TARGET_NAME}
+                        EXCLUDE_PATTERNS ${PROTO_SRCS} ${PROTO_HDRS})
+
+install(TARGETS ${TARGET_NAME} EXPORT ngraphTargets
+        RUNTIME DESTINATION ${NGRAPH_INSTALL_LIB} COMPONENT ngraph
+        ARCHIVE DESTINATION ${NGRAPH_INSTALL_LIB} COMPONENT ngraph
+        LIBRARY DESTINATION ${NGRAPH_INSTALL_LIB} COMPONENT ngraph)
```

Comment:
plugins don't need to install `ARCHIVE`, see https://github.com/openvinotoolkit/openvino/blob/master/cmake/developer_package/plugins/plugins.cmake#L125-L128

Reply:
Frontend is made as shared library, it is needed for integration with frameworks where concrete classes will be used